community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -118,7 +131,12 @@ public static Arguments buildArgs()
         return new Arguments()
                 .withArgument( new OptionalNamedArg( ARG_MEMORY, memory, memory,
                         "Recommend memory settings with respect to the given amount of memory, " +
-                        "instead of the total memory of the system running the command." ) );
+                        "instead of the total memory of the system running the command." ) )
+                .withArgument( new OptionalBooleanArg( ARG_DB, false,
+                        "Specific database to calculate page cache memory requirement for. " +
+                        "The generic calculation is still a good generic recommendation for this machine, " +
+                        "but there will be an additional calculation for minimal required page cache memory " +
+                        "for mapping all store and index files that are managed by the page cache." ) );
```

Comment:
I think the standard way of instructing a command to work on a database, is to use the `withDatabase` argument. I think we should do that here as well, instead of assuming that the configured default database is the one we want to build a configuration for in the particular invocation of the command.

Unfortunately, `withDatabase` does not allow you to override the description. I think that should be added, because the description you have here is really good.

Reply:
The `ConsistencyCheckCommand.execute` method has an example of using `arguments.has( "database" )` to check whether the `--database` argument was specified on the command line or not.